### ISSUE TITLE
Invalid `pointless-string-statement` for docstrings for Python 3.12 type-aliases (PEP 695)

### ISSUE THREAD

**Issue by Stausssi** [pylint-dev/pylint#9268]
### Bug description

```python
Adding a multiline docstring to a Python 3.12 type-alias (PEP 695) causing pylint to faild with a `pointless-string-statement`.

Example code:

type _GeneratorCallable = Callable[..., list[str]]
"""
Multiline
docstring
for a new type-alias
introduced in Python3.12
"""


Adding a `:` before the string also doesn't help:
```python
type _GeneratorCallable = Callable[..., list[str]]
""":
Multiline
docstring with :
for a new type-alias
introduced in Python3.12
"""
```
```


### Configuration

_No response_

### Command used

```shell
pylint src
```

### Pylint output

```python
************* Module src.generator.generator
src/generator/generator.py:8:0: W0105: String statement has no effect (pointless-string-statement)
```

### Expected behavior

Pylint should not raise a warning here

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.12.0 (main, Nov  2 2023, 13:03:57) [Clang 15.0.0 (clang-1500.0.40.1)]
```

### OS / Environment

macOS Sonoma, iTerm2

**Comment by jacobtylerwalls** [comment]
Ah. Here's this, if helpful:
```py
>>> n = ast.parse("""
... type MyTuple = tuple[str, str]
... "my docstring"
... """)
>>> n.body[0].__doc__
'TypeAlias(expr name, type_param* type_params, expr value)'
```

> We exclude Assign and AnnAssign because of this rejected [PEP](https://peps.python.org/pep-0224/), but for this case I don't see that?

Interesting, I would have thought we just were being friendly to (someone's) common practice (somewhere), not observing a rejected PEP.

**Comment by DanielNoord** [comment]
I think the original rationale was that PEP.

I think we might actually not want to do this? As this makes it clearer that the statement is indeed pointless and does not do what you think it does?

**Comment by jacobtylerwalls** [comment]
Yeah, I find that convincing.

**Comment by Stausssi** [comment]
Good to know that this type of docstring is actually **not** the way to go. Been using it for a while and all my tools picked it up fine so far.

**Comment by Stausssi** [comment]
I figured VS Code not showing any kind of hover information is just an "early adopter" problem. Was referring to that type of docstring for attributes in my earlier comment.
In fact, I can't seem to get any type of docstring format to work for the new type alias. Seems like some sort of oversight in the PEP?

**Comment by jacobtylerwalls** [comment]
Maybe worth asking on discuss.python.org?

**Comment by Stausssi** [comment]
Feedback from python.org is that the new type aliases are to be treated the same as regular assignments regarding docstrings. In this case I suppose it means also excluding it like `Assign` and `AnnAssign`.

**Comment by Stausssi** [comment]
Thank you!
